src/coreComponents/linearAlgebra/interfaces/hypre/HypreMGR.cpp

```diff
@@ -58,6 +58,7 @@ void hypre::mgr::createMGR( LinearSolverParameters const & params,
   GEOS_LAI_CHECK_ERROR( HYPRE_MGRSetTol( precond.ptr, 0.0 ) );
   GEOS_LAI_CHECK_ERROR( HYPRE_MGRSetMaxIter( precond.ptr, 1 ) );
   HYPRE_Int logLevel = LvArray::math::max( LvArray::integerConversion< HYPRE_Int >( params.logLevel - 1 ), LvArray::integerConversion< HYPRE_Int >( 0 ) );
+  logLevel &= ~0x2;
```

Comment:
could you please leave a comment about what is going on here?

Reply:
Sure, done in [fb95b52](https://github.com/GEOS-DEV/GEOS/pull/2928/commits/fb95b52129ab8fecb1f7569f81250e15c6eb4f74) Is it clear now?

Reply:
not really, even more confusing :disappointed:
maybe just provide a summary of possible logLevel values and what they do?

Reply:
The summary idea is good. @castelletto1 where should we put it?

Why did you find it more confusing? serious question that will help me to come up with a better explanation

Reply:
i am just not sure how print levels work on hyore side, so
`Disable preconditioner application info (solve phase)`
does not tell much and not sure why it needs to be disabled?

Reply:
Here's how it works for hypre: https://github.com/hypre-space/hypre/blob/57bfb26e268ddf003668c5d0b5938ae258922a83/src/parcsr_ls/HYPRE_parcsr_ls.h#L4285-L4311

Option 2 (second bit, aka. 0x2) turns on convergence history for the MGR solver. But here, we use MGR as a preconditioner to a Krylov method (e.g., gmres), and what we really want is to see is the convergence history of the solver, not MGR's. Therefore, I'm explicitly turning off the second bit (0x2) in this line.

With this info in mind, would you guys help me coming up with a better comment to the purpose of this line?

Reply:
> Option 2 (second bit, aka. 0x2) turns on convergence history for the MGR solver. But here, we use MGR as a preconditioner to a Krylov method (e.g., gmres), and what we really want to see is the convergence history of the krylov method, not MGR's. Therefore, I'm explicitly turning off the second bit (0x2) in this line.

This comment is good :)

Reply:
Great, I tweaked a bit and updated this in [7ee30a3](https://github.com/GEOS-DEV/GEOS/pull/2928/commits/7ee30a3d9fb6496ec0ca144dac75e332eae5b474)